retrieve_files.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+#set -x
+
+###################################################################################################################
+#
+#Script to download,extract, and parse PTAB pdf files from USPTO bulk data site. 
+#
+#argument options:
+#           (1) pass -d OR --date AND YYYYMMDD to download files from a specific date on
+#           (2) pass -a OR --all to download all files
+#           (3) pass -n OR --none to skip downloading of files and execute unzipping process
+#
+###################################################################################################################
+
+#create lock file
+function lock ( ) {
+  if "$1" == true
+  then
+    if [ -f ${LOCKFILE} ]
+    then
+      echo
+      echo "${LOCKFILE} exits."
+      echo "Looks like a copy of $scriptName is already running... "
+      echo "Abort"
+      log "ERR" " Already running, please check logs for reason... "
+      exit 0
+    else
+      touch ${LOCKFILE}
+    fi
+  else
+    if [ -f $LOCKFILE ] ; then
+      rm $LOCKFILE
+    fi
+  fi
+}
+
+function usage
+{
+    echo "usage: ./retrieve_files.sh [[-d YYYMMDD]| [-a] | [-n]]"
+}
+
+function log()
+{
+  type=$1
+  message=$2
+
+  if [ $type = "WARN" -o $type = "ERR" ]
+  then
+    echo >> $statusDirectory/err-$processingTime
+    echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message" >> $statusDirectory/err-$processingTime
+  fi
+  #
+  # always write into log file
+  #
+  echo
+  echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message"
+  echo >> $statusDirectory/log-$processingTime
+  echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message" >> $statusDirectory/log-$processingTime
+}
+
+#=============================== MAIN BODY OF SCRIPT ===============================
+
+##### Constants
+processingTime=`date +%Y%m%d-%H%M%S`
+scriptName=$0
+statusDirectory=logs
+BASEURL="https://bulkdata.uspto.gov/data2/patent/trial/appeal/board/"
+DROP_LOCATION="files/`date +%Y%m%d`"
+declare -i filedate
+retrieve_all=false
+retrieve_none=false
```

Comment:
retrieve_none is to control whether you want to skip the file download portion of the script and move on to the unzip process.  Eventually, I'd like to change the script so each processing part (download, uncompress, parse) is optional depending on the arguments passed in to the script.